recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt

```diff
@@ -0,0 +1,131 @@
+package com.mapbox.vision.recorder
+
+import android.annotation.SuppressLint
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.os.Environment
+import android.text.Html
+import android.view.View
+import android.view.WindowManager
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.mapbox.vision.VisionManager
+import com.mapbox.vision.common.BaseActivity
+import com.mapbox.vision.common.view.hide
+import com.mapbox.vision.common.view.show
+import com.mapbox.vision.mobile.core.interfaces.VisionEventsListener
+import com.mapbox.vision.mobile.core.models.Camera
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils
+import com.mapbox.vision.mobile.core.utils.snapdragon.SupportedSnapdragonBoards
+import com.mapbox.vision.performance.ModelPerformance
+import com.mapbox.vision.performance.ModelPerformanceConfig
+import com.mapbox.vision.utils.VisionLogger
+import kotlinx.android.synthetic.main.activity_recording.*
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+class RecordingActivity : BaseActivity() {
+
+    companion object {
+        private val BASE_SESSION_PATH = "${Environment.getExternalStorageDirectory().absolutePath}/Telemetry"
+        private val dateFormat = SimpleDateFormat("yyyy-MM-dd_HH-mm-ssZ", Locale.US)
+    }
+
+    private var isPermissionsGranted = false
+    private var visionManagerWasInit = false
+
+    private var appModelPerformanceConfig: ModelPerformanceConfig = ModelPerformanceConfig.Merged(
+        performance = ModelPerformance.Off
+    )
+
+    private val visionEventsListener = object : VisionEventsListener {
+
+        override fun onCountryUpdated(country: Country) {
+            println("Country $country")
+        }
+
+        override fun onCameraUpdated(camera: Camera) {
+            runOnUiThread {
+                fps_performance_view.setCalibrationProgress(camera.calibrationProgress)
+            }
+        }
+
+        @SuppressLint("SetTextI18n")
+        override fun onUpdateCompleted() {
+            val frameStatistics = VisionManager.getFrameStatistics()
+            runOnUiThread {
+                fps_performance_view.showInfo(frameStatistics, appModelPerformanceConfig)
+            }
+        }
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_recording)
+    }
+
+    public override fun onPermissionsGranted() {
```

Comment:
Is `public` needed here?

Reply:
yep, a base class is stored in a separate module